src/datasets/utils/py_utils.py

```diff
@@ -515,19 +514,6 @@ class Pickler(dill.Pickler):
 
     dispatch = dill._dill.MetaCatchingDict(dill.Pickler.dispatch.copy())
 
-    def save_global(self, obj, name=None):
-        if sys.version_info[:2] < (3, 7) and _CloudPickleTypeHintFix._is_parametrized_type_hint(
```

Comment:
I think you can also remove the `_CloudPickleTypeHintFix` class defined a bit later in this file